tracing/src/main/java/com/palantir/tracing/Tracer.java

```diff
@@ -91,10 +96,11 @@ public static void initTrace(Optional<Boolean> isObservable, String traceId) {
      */
     public static OpenSpan startSpan(String operation, String parentSpanId, SpanType type) {
         Trace current = getOrCreateCurrentTrace();
-        Preconditions.checkState(current.isEmpty(),
+        checkState(current.isEmpty(),
                 "Cannot start a span with explicit parent if the current thread's trace is non-empty");
-        Preconditions.checkArgument(parentSpanId != null && !parentSpanId.isEmpty(),
-                "parentTraceId must be non-empty: %s", parentSpanId);
+        checkArgument(parentSpanId != null && !parentSpanId.isEmpty(),
+                "parentSpanId must be non-empty",
+                SafeArg.of("parentSpanId", parentSpanId));
```

Comment:
nit: Since this is a very hot code path, I'd prefer to rewrite checks with args to this form, avoiding Arg creation:

```java
if (parentSpanId == null || parentSpanId.isEmpty()) {
    throw new SafeIllegalArgumentException("parentSpanId must be non-empty",
                SafeArg.of("parentSpanId", parentSpanId));
}
```

Reply:
Per your comment above, this arg isn't very helpful so we can probably just remove it an eliminate most cost here.